pkg/sql/codegen/pai/explain_render.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pai
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/aliyun/aliyun-oss-go-sdk/oss"
+)
+
+// ExplainRender used for PAI
+type ExplainRender struct {
+	Code         string
+	PaiCmd       string
+	Requirements string
+	// the explain result saved on OSS
+	endpoint string
+	ak       string
+	sk       string
+	bucket   string
+	key      string
+}
+
+func newExplainRender(userID string) *ExplainRender {
+	k := fmt.Sprintf("explain_images/%s/%s", userID, time.Now().Format("20060102150405"))
+	return &ExplainRender{
+		endpoint: os.Getenv("SQLFLOW_OSS_ALISA_ENDPOINT"),
+		ak:       os.Getenv("SQLFLOW_OSS_AK"),
+		sk:       os.Getenv("SQLFLOW_OSS_SK"),
+		bucket:   os.Getenv("SQLFLOW_OSS_ALISA_BUCKET"),
+		key:      k}
+}
+
+// Draw prints the explain result(png) to stdout
+func (expn *ExplainRender) Draw() error {
```

Comment:
Can pass in the OSS credential to `Draw` function, so that we don't need  `ak/sk/endpint...` in the ExplainRender struct.

Reply:
Not only the `Draw()` needs the credential but also the Python code in `pai/template_xgboost.go` needs. I prefer keeping such credential in one place
What's more, I'm going to use GFile to eliminate `ak/sk`.  #1872

Reply:
I think gfile is fine.